src/components/Dropdown/Dropdown.tsx

```diff
@@ -89,15 +89,17 @@ export class Dropdown extends React.Component<DropdownProps, State> {
   }
 
   static getDerivedStateFromProps(nextProps, state) {
+    if (state.selectedOption && nextProps.initialSelectedId !== state.selectedOption.id || !state.selectedOption) {
+      return {
+        selectedOption:
+          nextProps.options.find(
+            option => option.id === nextProps.initialSelectedId,
+          ) || null,
+      };
+    }
     if (state.selectedOption) {
       return null;
     }
```

Comment:
In this case, we should always return null, and not only if `state.selectedOption`